### DISCUSSION TITLE
从umi2.x升级到3.x 总报错‘Validate config "routes" failed, "value" must be an array’

### DISCUSSION THREAD

**Discussion by cxw2014** [umijs/umi#12147] [Q&A]
从umi2.x升级到3.x ，执行build 总报错‘Validate config "routes" failed, "value" must be an array’

**Comment by fz6m** [comment]
重新配置一下路由表吧，umi 3 和 umi 4 的 `routes` 配置都是数组，你可能配置成了一个不符合预期结构的对象，详情参考文档。

另外 umi 3 已经不维护了，建议升级到 umi 4 。
